rust/lance-table/src/rowids/segment.rs

```diff
@@ -392,6 +392,62 @@ impl U64Segment {
         let stats = Self::compute_stats(make_new_iter());
         Self::from_stats_and_sequence(stats, make_new_iter())
     }
+
+    pub fn mask(&mut self, positions: &[u32]) {
+        if positions.is_empty() {
+            return;
+        }
+        if positions.len() == self.len() {
+            *self = Self::Range(0..0);
+            return;
+        }
+        let count = (self.len() - positions.len()) as u64;
+        let sorted = match self {
+            Self::Range(_) => true,
+            Self::RangeWithHoles { .. } => true,
+            Self::RangeWithBitmap { .. } => true,
+            Self::SortedArray(_) => true,
+            Self::Array(_) => false,
+        };
+        // To get minimum, need to find the first value that is not masked.
+        let first_unmasked = (0..self.len())
+            .zip(positions.iter().cycle())
+            .filter(|(sequential_i, i)| **i != *sequential_i as u32)
+            .map(|(sequential_i, _)| sequential_i)
+            .next()
```

Comment:
Can you use [`find`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find) or `find_map`?